### PR TITLE
feat(toast)!: migrate toast to core tokens

### PR DESCRIPTION
**Pull request by yosevu**
## Description

This PR migrates the Toast component to core-tokens.

- Removes deprecated variants (warning, error, success)

## How and where has this been tested?

- Tested locally referencing the Components batch 2 XD file.
- Tested with VoiceOver on Safari
- Tested WHCM with forced colors in Chrome Emulation

## To-do list
~~- [ ] If my change impacts other components, I have tested to make sure they don't break.~~
~~- [ ] If my change impacts documentation, I have updated the documentation accordingly.~~
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] I have tested these changes in Windows High Contrast mode.

- [x] This pull request is ready to merge.

### CODE REVIEW

File: components/toast/index.css [hunk 1]
```diff
@@ -86,12 +184,15 @@ governing permissions and limitations under the License.
   align-self: center;
 
   .spectrum-Button {
+    /* https://caniuse.com/mdn-css_properties_float_flow_relative_values */
     float: inline-end;
```
Comment (yosevu) [inline components/toast/index.css:188]: This only works on Firefox.  Looking for a better supported approach.

File: components/toast/index.css [hunk 2]
```diff
@@ -30,54 +79,102 @@ governing permissions and limitations under the License.
...
 
-  font-size: var(--spectrum-toast-info-text-size);
-  font-weight: var(--spectrum-toast-info-text-font-weight);
-  line-height: var(--spectrum-toast-info-text-line-height);
+  line-height: var(--mod-toast-line-height, var(--spectrum-toast-line-height));
+
+  &:lang(js),
+  &:lang(zh),
+  &:lang(ko) {
+    line-height: var(--spectrum-toast-line-height-CJK, var(--mod-toast-line-height, var(--spectrum-toast-line-height)));
```
Comment (yosevu) [inline components/toast/index.css:157]: I changed the token  naming format `--system-component-style-description-modifier` to try to be consistent. Are there any issues with this approach?

Comment (pfulton) [reply components/toast/index.css:152]: I think this is fine. Thanks for the heads up!

File: components/toast/index.css (hunk 1 above)
Comment (yosevu) [reply components/toast/index.css:183]: `float: inline-end` and `:dir(ltr) { float: right; }` are the only approaches I found, but both are Firefox only. Should I have `float: right` as a fallback even though it won't work for RTL direction @pfulton?

File: components/toast/index.css [hunk 3]
```diff
@@ -10,18 +10,60 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+/* toast/index.css
```
Comment (bernhard-adobe) [inline components/toast/index.css:13]: Maybe remove this code if we don't need it anymore

File: components/toast/index.css [hunk 4]
```diff
@@ -10,18 +10,60 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
-  );
-  --spectrum-toast-content-padding-bottom: var(
-    --spectrum-global-dimension-size-65
-  );
-
-  --spectrum-toast-button-margin-right: var(
-    --spectrum-global-dimension-size-130
-  );
+  /* Hardcoded variables */
+  --spectrum-toast-font-weight: 400;
```
Comment (bernhard-adobe) [inline components/toast/index.css:26]: maybe you could use this custom-var for now:
https://github.com/adobe/spectrum-css/blob/main/components/tokens/custom-spectrum/custom-vars.css#L32
`  --spectrum-font-weight-regular: 400;`

File: components/toast/index.css [hunk 5]
```diff
@@ -10,18 +10,60 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+
+  /* Size */
+
+  --spectrum-toast-font-size: var(--spectrum-font-size-100);
```
Comment (bernhard-adobe) [inline components/toast/index.css:30]: I think in all of these definitions we need to add the `--mod`  option as well to enable custom overwrites. 

like `  --spectrum-toast-font-size: var(--mod-font-size-100, var(--spectrum-font-size-100);
`

more details here: https://github.com/adobe/spectrum-css/wiki/Migrating-a-component-to-core-tokens#implementing-the-custom-properties-api

File: components/toast/index.css [hunk 6]
```diff
@@ -10,18 +10,60 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+  --spectrum-toast-corner-radius: var(--spectrum-corner-radius-100);
+
+  --spectrum-toast-line-height: var(--spectrum-line-height-100);
+  --spectrum-toast-line-height-CJK: var(--spectrum-CJK-line-height-100);
```
Comment (bernhard-adobe) [inline components/toast/index.css:34]: We may want to use for CJK support the lang-selector:
```CSS
 &:lang(ja),
  &:lang(zh),
  &:lang(ko) {
    line-height: var(--mod-toast-line-height-CJK, var(----spectrum-toast-line-height-CJK));
  }

File: components/toast/index.css [hunk 7]
```diff
@@ -10,18 +10,60 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+  --spectrum-toast-negative-background-color-default: var(--spectrum-negative-background-color-default);
+  --spectrum-toast-positive-background-color-default: var(--spectrum-positive-background-color-default);
+  --spectrum-toast-informative-background-color-default: var(--spectrum-informative-background-color-default);
+
+  --spectrum-toast-text-and-icon-color: var(--spectrum-white);
+}
+
+@media (forced-colors: active) {
+  .spectrum-Toast {
+    border: 1px solid transparent;
```
Comment (bernhard-adobe) [inline components/toast/index.css:65]: Maybe we could soft-code this `1px` value? A bit of a nit, not a bit priority or blocker

File: components/toast/index.css [hunk 8]
```diff
@@ -30,68 +72,120 @@ governing permissions and limitations under the License.
...
 
-  font-size: var(--spectrum-toast-neutral-text-size);
-  font-weight: var(--spectrum-toast-neutral-text-font-weight);
   -webkit-font-smoothing: antialiased;
+
+  background-color: var(--mod-toast-background-color-default, var(--spectrum-toast-background-color-default));
+  color: var(--mod-toast-background-color-default, var(--spectrum-toast-background-color-default));
+
+  &--negative,
+    /* @deprecated */.spectrum-Toast--error {
```
Comment (bernhard-adobe) [inline components/toast/index.css:88]: remove if deprecated?

File: components/toast/index.css [hunk 9]
```diff
@@ -30,68 +72,120 @@ governing permissions and limitations under the License.
...
+
+    .spectrum-Toast-closeButton {
+      &:focus-ring:not(:active) {
+        color: var(--mod-toast-informative-background-color-default, var(--spectrum-toast-informative-background-color-default));
+      }
+    }
+  }
+
+  &--positive,
+    /* @deprecated */.spectrum-Toast--success {
```
Comment (bernhard-adobe) [inline components/toast/index.css:111]: remove if deprecated?

File: components/toast/index.css (hunk 1 above)
Comment (bernhard-adobe) [reply components/toast/index.css:183]: Can you use a browser-non specific version?

**Review by bernhard-adobe — CHANGES_REQUESTED** [review]
a couple things that we may want to work on. Mostly the `--mod` custom modifier that seems to e missing from a couple of values.

File: components/toast/index.css (hunk 1 above)
Comment (yosevu) [reply components/toast/index.css:183]: Hey Bernhard, both options that I have found are only ones I found that are based on directionality. `float: right;` is used currently. I haven't found another approach.

File: components/toast/index.css (hunk 5 above)
Comment (yosevu) [reply components/toast/index.css:30]: Oh I added the `--mod` prefix when I use the component token e.g. `font-size: var(--mod-toast-font-size, var(--spectrum-toast-font-size));`. Is that what you mean?

File: components/toast/index.css (hunk 6 above)
Comment (yosevu) [reply components/toast/index.css:34]: I used the lang selector below in the `.spectrum-Toast-content` styles. Is that what you meant?

File: components/toast/index.css (hunk 7 above)
[the remark this answers is not in the document]
Comment (yosevu) [reply components/toast/index.css:65]: Updated.

File: components/toast/index.css [hunk 10]
```diff
@@ -30,68 +72,120 @@ governing permissions and limitations under the License.
...
-  .spectrum-Button,
-  .spectrum-ClearButton {
-    & + .spectrum-Button,
-    & + .spectrum-ClearButton {
-      margin-inline-start: var(--spectrum-toast-neutral-button-gap-x);
-    }
+  margin-block-start: var(--mod-toast-spacing-top-edge-to-divider, var(--spectrum-toast-spacing-top-edge-to-divider));
+  margin-block-end: var(--mod-toast-spacing-bottom-edge-to-divider, var(--spectrum-toast-spacing-bottom-edge-to-divider));
+
+  border-inline-start-color: rgba(255, 255, 255, 0.2);
```
[the remark this answers is not in the document]
Comment (yosevu) [reply components/toast/index.css:166]: Yeah, makes sense to use it base don that note. Thank you.

File: components/toast/index.css (hunk 5 above)
Comment (bernhard-adobe) [reply components/toast/index.css:19]: yes, just checking this is consistently used.

File: components/toast/index.css (hunk 6 above)
Comment (bernhard-adobe) [reply components/toast/index.css:24]: maybe we should used that when referencing CJK line-heights? @pfulton what do you suggest?

File: components/toast/index.css (hunk 7 above)
Comment (pfulton) [reply components/toast/index.css]: Can we use the custom property defined [here](https://github.com/adobe/spectrum-css/pull/1483/files#diff-dd286afa975e8faf641af07455647fd4d891bcc2997ac531249fa23a6dad2b07R65)?

```css
border: var(--mod-toast-border-width, var(--spectrum-toast-border-width)) solid transparent;
```

File: components/toast/index.css (hunk 8 above)
Comment (pfulton) [reply components/toast/index.css]: This is a great question that we should maybe surface to design and/or Jess. I'm inclined to think that we should only show current guidance and get rid of the "use X variant instead" examples in the docs.

Comment (pfulton) [reply components/toast/index.css]: Asked [here](https://adobedesign.slack.com/archives/G019JTYMT6H/p1659387331617259)

File: components/toast/index.css (hunk 6 above)
Comment (pfulton) [reply components/toast/index.css:24]: @bernhard-adobe could you clarify? I think @yosevu implemented it as I'd expect [here](https://github.com/adobe/spectrum-css/blob/3e3807d310b77681cd07660e10bbd970aedfcfdd/components/toast/index.css#L138-L142).

@yosevu You may have a typo on line 138: `&:lang(js)` should be `&:lang(ja)`, right?

Also, can we lowercase `CJK` in the custom property name, please?

Comment (bernhard-adobe) [reply components/toast/index.css:24]: @pfulton could we lower-case CJK in core tokens name, please ;)?

Comment (yosevu) [reply components/toast/index.css]: Ah thanks. Updated in e3f4a568096fb55054c773f48d783c9a00994d2b

File: components/toast/index.css (hunk 7 above)
Comment (yosevu) [reply components/toast/index.css]: Yes. I think this is outdated. I refactored this to use that property.

File: components/toast/index.css (hunk 8 above)
Comment (yosevu) [reply components/toast/index.css]: I removed the deprecated variants.

**Comment by pfulton** [comment]
Beta released:
`- @spectrum-css/toast@8.0.0-beta.0`

**Comment by pfulton** [comment]
New beta released due to faulty build:
`- @spectrum-css/toast@8.0.0-beta.1`

**Comment by pfulton** [comment]
@yosevu We're seeing some size changes and close button issues when we try to integrate this with the Spectrum Web Components project.

[Here's my draft PR](https://github.com/adobe/spectrum-web-components/pull/2485) where you can take a look at their VRT results.

Would you mind double-checking on the sizing and maybe also try bringing in the latest release of the CloseButton? That might actually help things.

File: components/toast/index.css (hunk 1 above)
Comment (bernhard-adobe) [reply components/toast/index.css]: Awesome! thanks for checking! `float: right` it is!

**Review by bernhard-adobe — COMMENTED** [review]
Approved but won't hold clicking the approval button.  Still need to address however the comment by @pfulton about the impact on Spectrum Web Components

**Comment by yosevu** [comment]
I updated the token used for the `padding-inline` of the close button. I believe this was the issue based on the comment in @pfulton draft PR. Would you mind taking a look confirm it @bernhard-adobe? I also updated to the latest CloseButton package.

Design:
Screen Shot 2022-08-10 at 10 14 35 PM
Beta:
Screen Shot 2022-08-10 at 10 16 48 PM

Updated:
Screen Shot 2022-08-10 at 10 17 39 PM

**Comment by bernhard-adobe** [comment]
VRT is: https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/31/

**Comment by bernhard-adobe** [comment]
VRT test is 	https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/31/artifact/backstop_data/html_report/index.html
[2022-08-10 -- 1483 - toast.zip](https://github.com/adobe/spectrum-css/files/9306260/2022-08-10.--.1483.-.toast.zip)

**Comment by pfulton** [comment]
Released:
`8.0.0-beta.2`

**Review by bernhard-adobe — APPROVED** [review]
released as beta so I would approve this PR ;)

**Comment by yosevu** [comment]
Could I get a new release for this @pfulton  or @GarthDB?

**Comment by pfulton** [comment]
Released:
`@spectrum-css/toast@8.0.0-beta.3`

**Comment by pfulton** [comment]
I think we're ready to merge this in and graduate the release.

**Pull request merged**